microcosm_caching/decorators.py

```diff
@@ -78,21 +86,21 @@ def retrieve_from_cache(key: str):
 
         return resource
 
-    def set_in_cache(key: str, value: Any) -> None:
+    def add_in_cache(key: str, value: Any) -> None:
         start_time = perf_counter()
 
-        resource_cache.set(key, value, ttl=ttl)
+        resource_cache.add(key, value, ttl=ttl)
```

Comment:
Add => set if not present.

Reply:
can we actually use cas here?

Reply:
I was trying it earlier, I think that solves a slightly different problem: race between gets and sets. I wanted to make sure the race between delete and set is solved.

From what I could tell, cas is used for when you want to update a key, but not if others have updated in the interim (hence the token on retrieve). I preferred to just delete the key wholesale. I could be misunderstanding, though.

Reply:
i see. add will keep it at the top of the lru

Reply:
Is that a problem?

Reply:
no... I should say the bottom of the lru. "add" bumps the frequency for which the item is requested, preventing it from being garbage collection. this is fine as is.